### PR TITLE
feat: add Anthropic models support

### PR DESCRIPTION
**Pull request by danilodiez**
Hi team, I've been working on supporting Anthropic and Claude models into guidance: https://github.com/microsoft/guidance/issues/50

image

Please let me know if you have any feedback 😄

### CODE REVIEW

**Comment by danilodiez** [comment]
@microsoft-github-policy-service agree company="LenioLabs"

**Review by plusInfinity — COMMENTED** [review]
Have you forgotten to include `_anthropic.py` in your PR request?

**Comment by danilodiez** [comment]
> Have you forgotten to include `_anthropic.py` in your PR?

I actually did 😅 , updated!

**Comment by danilodiez** [comment]
Hi there @slundberg could you review this PR please? 😄

**Comment by adamgordonbell** [comment]
Does this support claude2?

**Comment by danilodiez** [comment]
> 

I'll test and get back with an answer

**Comment by danilodiez** [comment]
Hi there @adamgordonbell, we now support Claude2 😄

File: setup.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ def find_version(*file_paths):
         "msal",
         "requests",
         "numpy",
+        "anthropic"
```
Comment (murphp15) [inline setup.py:43]: you are missing a comma at the end of this line

File: guidance/llms/_anthropic.py [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+        if self.api_type is not None:
+            anthropic.api_type = self.api_type
+        if self.api_version is not None:
+            anthropic.api_version = self.api_version
+        if self.api_base is not None:
+            anthropic.api_base = self.api_base
+
+        if self.chat_mode:
+            # kwargs['messages'] = prompt_to_messages(kwargs['prompt'])
+            kwargs['max_tokens_to_sample'] = 500
```
Comment (murphp15) [inline guidance/llms/_anthropic.py:305]: why is this hardcoded at 500?

File: setup.py (hunk 1 above)
Comment (danilodiez) [reply setup.py]: Fixed 😄

**Comment by BedirT** [comment]
Is there any update on this?

**Comment by josephpconley** [comment]
Is the only blocker here to resolve conflicts?

**Comment by anydote** [comment]
Great job! I'm eagerly waiting for this to be merged after the conflicts have been resolved.

**Comment by Harsha-Nori** [comment]
We've been through a major redesign in the interim, but just wanted to update here and say that `guidance.models.Anthropic` should work with all of the latest anthropic models :). Closing this PR for now given the massive codebase change, but feel free to open new issues/PRs if the new code has any troubles

**Pull request closed without merging**